File: backend/src/internal/models/bulletin.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Bulletin struct {
	ID uint `json:"id" gorm:"primaryKey"`

	Type string `json:"type" gorm:"size:100"`

	Title        string     `json:"title" gorm:"size:500" validate:"required,min=2"`
	Category     *uint      `json:"category,omitempty"`
	NickName     string     `json:"nickname,omitempty"`
	Description  string     `json:"description" gorm:"type:text"`
	Recommend    uint       `json:"recommend,omitempty"`
	NotRecommend uint       `json:"notrecommend,omitempty"`
	UserID       uint       `json:"userId"`
	Level        *uint      `json:"level,omitempty"`
	Status       bool       `json:"status"`
	Allas        *uint      `json:"allas,omitempty"`
	Top          bool       `json:"top" gorm:"default:false"`
	Check        uint       `json:"check"`
	DomainID     *uint      `json:"domainId" gorm:"index"`
	Domain       *Domain    `json:"domain" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Views        uint       `json:"views"`
	CreatedDate  *time.Time `json:"createdDate,omitempty"`

	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

File: frontend/src/src/actions/bulletin.ts
import { gql } from "@apollo/client";

export const GET_LATEST_BULLETINS = gql`
  query GetLatestBulletins {
    bulletins {
      id
      title
      category
      nickname
      description
      recommend
      notrecommend
      level
      status
      memberId
      alllas
      top
      check
      domainId
      domain {
        id
        name
      }
      userId
      user {
        id
        name
        userid
      }
      createdAt
      updatedAt
      deletedAt
    }
  }
`;
export const GET_BULLETINS = gql`
  query GetBulletins(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: getBulletins(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      bulletins {
        id
        title
        category
        nickname
        description
        recommend
        memberId
        notrecommend
        level
        status
        alllas
        top
        check
        domainId
        domain {
          id
          name
        }
        userId
        user {
          id
          name
          userid
        }
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const CREATE_BULLETIN = gql`
  mutation NewBulletin($input: NewBulletinInput!) {
    response: createBulletin(input: $input) {
      id
      title
      description
      category
      nickname
      recommend
      notrecommend
      level
      alllas
      memberId
      top
      check
      domainId
      domain {
        id
        name
      }
    }
  }
`;

export const UPDATE_BULLETIN = gql`
  mutation UpdateBulletin($id: ID!, $input: UpdateBulletinInput!) {
    response: updateBulletin(id: $id, input: $input) {
      id
      title
      description
      category
      nickname
      recommend
      notrecommend
      level
      alllas
      memberId
      top
      check
      domainId
      domain {
        id
        name
      }
      updatedAt
      createdAt
      deletedAt
    }
  }
`;

export const DELETE_BULLETIN = gql`
  mutation DeleteBulletin($id: ID!) {
    response: deleteBulletin(id: $id)
  }
`;